react/features/dial-out/components/DialOutNumbersForm.web.js

```diff
@@ -0,0 +1,319 @@
+        _dialOutCodes: React.PropTypes.array,
+
+        /**
+         * The function called on every dial input change.
+         */
+        onChange: React.PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func,
+
+        /**
+         * Invoked to send an ajax request for dial-out codes.
+         */
+        updateDialOutCodes: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new {@code DialOutNumbersForm} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            dialInput: '',
+
+            /**
+             * Whether or not the dropdown should be open.
+             *
+             * @type {boolean}
+             */
+            isDropdownOpen: false,
+
+            /**
+             * The selected country.
+             *
+             * @type {Object}
+             */
+            selectedCountry: DEFAULT_COUNTRY
+        };
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onInputChange = this._onInputChange.bind(this);
+        this._onOpenChange = this._onOpenChange.bind(this);
+        this._onSelect = this._onSelect.bind(this);
+    }
+
+    /**
+     * Dispatches a request for dial out codes if not already present in the
+     * redux store. If dial out codes are present, sets a default code to
+     * display in the dropdown trigger.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentDidMount() {
+        const dialOutCodes = this.props._dialOutCodes;
+
+        if (dialOutCodes) {
+            this._setDefaultCode(dialOutCodes);
+        } else {
+            this.props.updateDialOutCodes();
+        }
+    }
+
+    /**
+     * Monitors for dial out code updates and sets a default code to display in
+     * the dropdown trigger if not already set.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.selectedCountry && nextProps._dialOutCodes) {
+            this._setDefaultCode(nextProps._dialOutCodes);
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t, _dialOutCodes } = this.props;
+
+        const items
+            = _dialOutCodes ? this._formatCountryCodes(_dialOutCodes) : [];
+
+        return (
+            <div className = 'form-control'>
+                { this._createDropdownMenu(items) }
+                <div className = 'dial-out-input'>
+                    <input
+                        className = 'input-control'
+                        label = ''
+                        onChange = { this._onInputChange }
+                        placeholder = { t('dialOut.enterPhone') } />
```

Comment:
It didn't fit very well the styling of the dropbox. I was planning to have a second look once this first version is merged.